### DISCUSSION TITLE
what is react native

### DISCUSSION THREAD

**Discussion by ThushanVoxitec** [community/community#64751] [Mobile]
### Select Topic Area

Question

### Body

what is react native

**Comment by thushanvithana** [comment]
React Native is an open-source UI software framework created by Meta Platforms, Inc. It is used to develop applications for Android, Android TV, iOS, macOS, tvOS, Web, Windows and UWP by enabling developers to use the React framework along with native platform capabilities

**Comment by iqbal-web** [comment]
React Native is an exciting framework that enables web developers to create robust mobile applications using their existing JavaScript knowledge. It offers faster mobile development, and more efficient code sharing across iOS, Android, and the Web, without sacrificing the end user's experience or application quality.

Check more details ---> https://reactnative.dev/

**Comment by HimanshuGupta-p1** [comment]
React Native is an open-source JavaScript framework, designed for building apps on multiple platforms like iOS, Android, and also web applications, utilizing the very same code base. It is based on React, and it brings all its glory to mobile app development.

Both frameworks: ReactJS (web) and React Native were brought to life by Facebook. React Native was a Hackathon project aiming at solving the company’s biggest pain point - maintaining two code bases for their app. The problem with maintaining two code bases for such a big app? Work duplication and, at times, solving the same problem in two different ways. React Native is a straightforward answer to these problems.

React Native uses JavaScript to compile the app’s user interface, but uses native-OS views. For more complex features, it allows code implementation in OS-native languages (Swift and Objective-C for iOS, and Java and Kotlin for Android).

**Comment by krishvsoni** [comment]
React.js and React Native are both popular frameworks developed by Facebook for building user interfaces, but they are used for different purposes and have distinct characteristics. Your decision to learn one or the other should depend on your specific goals and the types of applications you want to develop. Here's a brief comparison, followed by my personal opinion:

React.js:

Purpose: React.js, also known as React or ReactJS, is a JavaScript library for building web applications. It's primarily used for creating user interfaces in web browsers.

Development Environment: React.js applications run in web browsers, which means you need a web browser to view and interact with them. You develop and test React.js applications in web development environments.

Platform: React.js is platform-agnostic and can be used to build web applications that run on any modern web browser.

Skills: Learning React.js involves mastering JavaScript and JSX (a syntax extension for JavaScript). It's an essential skill for web developers.

React Native:

Purpose: React Native is a framework for building mobile applications for iOS and Android using JavaScript and React principles. It allows you to build native mobile apps with a single codebase.

Development Environment: React Native applications run natively on mobile devices, so you need to set up development environments for both iOS and Android. You write code using JavaScript and React, but the user interface components are native.

Platform: React Native is specific to mobile app development. It's not used for web applications.

Skills: Learning React Native involves mastering JavaScript and React concepts, as well as understanding mobile app development for iOS and Android. It's a valuable skill if you want to build cross-platform mobile apps.

Personal Opinion:

My personal opinion is that the choice between React.js and React Native should be based on your specific goals and the types of projects you want to work on:

Learn React.js (React): Choose React.js if you are primarily interested in web development and want to become proficient in building web applications. React.js is a fundamental skill for web developers, and it provides a solid foundation for front-end development.

Learn React Native: Choose React Native if you are interested in mobile app development and want to create native mobile applications for both iOS and Android with a single codebase. React Native is a valuable skill if you want to focus on mobile app development or build cross-platform applications.

Ultimately, the choice depends on your career aspirations and project preferences. Some developers choose to learn both React.js and React Native to have a broader skill set and the ability to work on both web and mobile projects.
